site/premium.js.template

```diff
@@ -1,63 +1,84 @@
+            }
+            player.src = productDetails.signedUrl;
+            player.controls = true;
         } else {
-            throw new Error('Unsupported file type');
+            player = document.createElement('img');
+            player.src = productDetails.previewUrl;
+            player.addEventListener('click', async(event) => {
+                event.stopPropagation();
+                const loginResponse = await fetch(`${premiumServer}/login/${slug}`, {
+                    method: 'POST',
+                    headers: {'Content-Type': 'application/json'},
+                    body: JSON.stringify({authToken})
+                });
+                console.log(loginResponse);
+                if(loginResponse.status === 200) {
+                    window.open((await loginResponse.json()).url, '_blank');
+                }
+            });
         }
-        player.src = productDetails.signedUrl;
-        player.controls = true;
-    } else {
-        player = document.createElement('img');
-        player.src = productDetails.previewUrl;
-        player.addEventListener('click', () => {
-            window.open(productDetails.purchaseUrl, '_blank');
-        });
-    }
-    return player;
-};
+        return player;
+    };
 
-// Define Single SPA lifecycle entry points as AMD modules.
-define({
-    mount: async(sspaArgs) => {
+    // Show the player UI in the given domElement.
+    const show = async(domElement) => {
         const bbs = await getBbsCommon();
         const authToken = await bbs.getAuthToken();
-        const productDetails = await (await fetch(`${premiumServer}/productDetails?path=${filePath}`, {
+        const productDetails = await (await fetch(`${premiumServer}/product/${slug}`, {
             method: 'POST',
             headers: {'Content-Type': 'application/json'},
-            body: JSON.stringify({filePath, authToken})
+            body: JSON.stringify({authToken})
         })).json();
         const ui = document.createElement('div');
-        ui.appendChild(await makePlayer(productDetails));
-        sspaArgs.domElement.appendChild(ui);
-    },
-    // We are not currently using these lifecycle functions.
-    bootstrap: async() => {},
-    unmount: async() => {},
-    unload: async() => {}
-});
+        ui.appendChild(await makePlayer(productDetails, authToken));
+        domElement.appendChild(ui);
+    };
+
+    // Define Single SPA lifecycle entry points as AMD modules.
+    define({
+        mount: async(sspaArgs) => new Promise((resolve, reject) => {
+            try {
+                show(sspaArgs.domElement).then(resolve).catch(reject);
+            } catch(error) {
+                console.error(error);
+                sspaArgs.domElement.textContent = error.message;
+                reject(error);
+            }
+        }),
+        // We are not currently using these lifecycle functions but they are required by Single SPA.
+        bootstrap: async() => {},
```

Comment:
maybe add console.log("not supported");

Reply:
That would be misleading. It's not that `bootstrap` is not supported, it's just that we don't have anything meaningful to do on that event. The ordinary lifecycle of a SSPAA is:
1. Bootstrap - this gets called when the SSPAA is registered on a page
2. Mount - this gets called when there is a need to show the SSPAA
3. Unmount - this gets called when there is a need to hide the SSPAA
4. Unload - this gets called when there will be no further need to show the SSPAA, so everything can be deleted

But a plugin that renders posts is automatically shown ASAP, so there is no need for bootstrap (even though the event is called), it is never hidden, so there is no need for unmount, and it is never destroyed, no no unload. The only reason we have those stub functions is that without them, SSPA will throw an error upon registration.